indexer/feature_altitude.hpp

```diff
@@ -83,32 +86,43 @@ class Altitudes
   {
     CHECK(!m_altitudes.empty(), ());
 
-    WriteVarInt(sink, static_cast<int32_t>(m_altitudes[0]) - static_cast<int32_t>(minAltitude));
-    for (size_t i = 1; i < m_altitudes.size(); ++i)
+    BitWriter<TSink> bits(sink);
+    TAltitude prevAltitude = minAltitude;
+    for (auto const altitude : m_altitudes)
     {
-      WriteVarInt(sink,
-                  static_cast<int32_t>(m_altitudes[i]) - static_cast<int32_t>(m_altitudes[i - 1]));
+      CHECK_LESS_OR_EQUAL(minAltitude, altitude, ("A point altitude is less than min mwm altitude"));
+      uint32_t const delta = bits::ZigZagEncode(static_cast<int32_t>(altitude) -
+                                                static_cast<int32_t>(prevAltitude));
+      coding::DeltaCoder::Encode(bits, delta + 1 /* making it greater than zero */);
+      prevAltitude = altitude;
     }
   }
 
   template <class TSource>
   bool Deserialize(TAltitude minAltitude, size_t pointCount, TSource & src)
   {
-    m_altitudes.clear();
-    if (pointCount == 0)
-    {
-      ASSERT(false, ());
-      return false;
-    }
+    ASSERT_NOT_EQUAL(pointCount, 0, ());
 
-    m_altitudes.resize(pointCount);
+    BitReader<TSource> bits(src);
     TAltitude prevAltitude = minAltitude;
+    m_altitudes.resize(pointCount);
+
     for (size_t i = 0; i < pointCount; ++i)
     {
-      m_altitudes[i] = static_cast<TAltitude>(ReadVarInt<int32_t>(src) + prevAltitude);
+      uint32_t const biasedDelta = coding::DeltaCoder::Decode(bits);
+      if (biasedDelta == 0)
+      {
+        ASSERT(false, ("Decoded altitude delta is zero. Point number in its feature is", i));
+        m_altitudes.clear();
+        return false;
+      }
+      uint32_t const delta = biasedDelta - 1;
+
+      m_altitudes[i] = static_cast<TAltitude>(bits::ZigZagDecode(delta) + prevAltitude);
       if (m_altitudes[i] < minAltitude)
       {
-        ASSERT(false, ());
+        ASSERT(false, ("A point altitude read from file(", m_altitudes[i],
+                       ") is less then min mwm altitude(", minAltitude, "). Point number in its feature is", i));
```

Comment:
> So, maybe it worth to print feature id too?
>There is no useful debug information that 3-rd point of some feature was corrupted. But it would be helpful if we'll know that 3-rd point of, say, 1000nd feature in Russia_Moscow.mwm is corrupted.

I agree. Done.